test/Tests/src/Data/Numbers_Spec.enso

```diff
@@ -190,4 +194,10 @@ spec =
             1.ceil . should_equal 1
             almost_max_long_times_three_decimal.ceil.to_decimal . should_equal almost_max_long_times_three_plus_1.to_decimal
             almost_max_long_times_three_plus_1.ceil . should_equal almost_max_long_times_three_plus_1
-
+        Test.specify "should parse text as decimals" <|
+           result = [-1.0, 0.0, 0.0, 1.0, 2.0, 3.4, Nothing]
+           text_literals.map Decimal.parse . should_equal result
+    Test.group "Numbers" <|
+        Test.specify "should parse text as numbers" <|
+           result = [-1, 0, 0, 1.0, 2.0, 3.4, Nothing]
```

Comment:
This is very odd behaviour. If it is a valid integer it should be _parsed_ as a valid integer. The input list should result in `[-1, 0, 0, 1, 2, 3.4, Parse_Error]`. Basically, it should parse the numbers based on the natural representation for their value in our unified number system.